### PR TITLE
feat: item arrays in samples

### LINKED ISSUE #16: Feature Request: support arrays of same item definition in sample
For some simulations it can be useful to have an array of the same item definition.  For example a bar with ultrasonic sensors that measure the transverse profile or a roadway.  Such a bar could have 30 sensors spaced at 10cm intervals.  

Currently adding the same item definition in the config more than once, leads to only one instance of that item being created. 

e.g.

```
---
ttl: 10000
broker:
  host: 52.53.192.249
  port: 8883
  auth:
    username: tester
    password: changeit
  tls:
    trustStore: "./scripts/keys/brokerTrust.jks"
    trustPass: "ENCqTJZQarWDANjbiKQRH1R5/Dw3jNtSIYq12fIt67sIPEAAAAQmi6eCz/B3DynfmBHkC30s9n9/ynDhlcNo2yDA7ma90k="
items:
  - name: "transverse"
    type: "Double"
    label: "cm"
    max: 25.0
    min: 5.0
    period: 1
  - name: "verticalAccel"
    type: "Double"
    label: "gForce"
    max: 2.0
    min: -2.0
    period: 1
  - name: "speed"
    type: "Double"
    label: "kph"
    max: 120.0
    min: 0.0
    period: 7
samples:
  - id: "roadData"
    name: "roadData"
    topic: "test/rdata"
    items:
      - "transverse"
      - "transverse"
      - "transverse"
      - "transverse"
      - "transverse"
      - "verticalAccel"
      - "speed"
devices:
  - id: "roadSampler"
    name: "Road Sampler 01"
    description: "typical road sampler"
    interval: 1000
    jitter: 0
    count: 1
    samples:
      - "roadData"

```

### PR DESCRIPTION
**Pull request by karel-rehor**
Adds arrays of items to samples, including serialization style options. 

* Closes #16 
* Should close backlog items (https://github.com/orgs/bonitoo-io/projects/2/views/1)
   * Item Type reuse in YAML config
   * Issue 16 - support for arrays of items of same type in samples

### CODE REVIEW

File: src/main/java/io/bonitoo/qa/conf/data/ItemNumConfig.java
Comment (ivankudibal) [inline src/main/java/io/bonitoo/qa/conf/data/ItemNumConfig.java:1]: line 6,7 the import of java.util.Vector, java.util.Array are never used

File: src/main/java/io/bonitoo/qa/plugin/sample/SamplePlugin.java [hunk 1]
```diff
@@ -83,8 +86,29 @@ public SamplePlugin(PluginProperties props, SampleConfig config) {
     this.id = config.getId();
     this.topic = config.getTopic();
     this.items = new HashMap<>();
+    // TODO better support for arrays of items
```
Comment (ivankudibal) [inline src/main/java/io/bonitoo/qa/plugin/sample/SamplePlugin.java:89]: what kind of better support, or is this resolved?

**Review by ivankudibal — APPROVED** [review]
There is  itemArrayTest.yml in resources. Should we create an array example as well?

File: src/main/java/io/bonitoo/qa/conf/data/ItemNumConfig.java
Comment (ivankudibal) [reply src/main/java/io/bonitoo/qa/conf/data/ItemNumConfig.java:1]: this kind of problem occurs at more places and not relevant to this PR.. perhaps to be addressed in a separated issue

File: src/main/java/io/bonitoo/qa/plugin/sample/SamplePlugin.java (hunk 1 above)
Comment (karel-rehor) [reply src/main/java/io/bonitoo/qa/plugin/sample/SamplePlugin.java:89]: This was already addressed in changes below this line.  Forgot to remove todo comment.

File: src/main/java/io/bonitoo/qa/conf/data/ItemNumConfig.java
Comment (karel-rehor) [reply src/main/java/io/bonitoo/qa/conf/data/ItemNumConfig.java:1]: These lines are fixed.  Will search for other unused import statements.  Thought checkstyle caught them.  Perhaps not.

**Pull request merged**